api/src/test/java/org/openmrs/PatientTest.java

```diff
@@ -27,6 +28,29 @@
  */
 public class PatientTest {
 	
+	/**
+	 * Regression test for TRUNK-3118
+	 */
+	@Test
+	public void patient_shouldConstructCloneWhenPassedPatient() {
+		Patient p = new Patient();
+		PatientIdentifier pa1 = new PatientIdentifier();
+		PatientIdentifierType identifierType = new PatientIdentifierType(1);
+		Location location = new Location(1);
+
+		pa1.setIdentifier("theid");
+		pa1.setIdentifierType(identifierType);
+		pa1.setLocation(location);
+		pa1.setVoided(true);
+		p.addIdentifier(pa1);
+		p.setAllergyStatus("TestAllergy");
+		p.setId(1);
+		
+		Patient p2 = new Patient(p);
+
+		assertEquals(p, p2);
```

Comment:
@shaoyuancc the default equal implementation for equals for OpenMRS data just compares uuids (see BaseOpenMRSObject.equals()) so it would be good to add a few other tests to make sure that all the fields are copied over properly.